### ISSUE TITLE
Avoid modifying global state of other packages (dask) at import

### ISSUE THREAD

**Issue by pelson** [SciTools/iris#2739]
It is bad form to modify a package's global state when another package is imported.

We shouldn't be changing the dask options when ``iris._lazy_data`` is imported. If it is that important to us, we should have our own state that we modify and maintain.

Note: Cartopy does a [similar thing](https://github.com/SciTools/cartopy/blob/master/lib/cartopy/__init__.py#L26), and it is bad news.

**Comment by pelson** [comment]
Discussion with @dkillick. Pseudo code of one potential approach:

```
iris.config.get

    if iris.config._da_getter is explicitly set
        use it
    else
        get da.config
            if the bad one
                change to the default iris one
            else
                use da.config
```

**Comment by pelson** [comment]
Closed by #2879
